### DISCUSSION TITLE
Possible for Bash scripts to return JSON?

### DISCUSSION THREAD

**Discussion by skurfuerst** [windmill-labs/windmill#1612] [Questions] [answered]
Hey,

I am currently trying to write a bash script which should return a PNG via the https://docs.windmill.dev/docs/reference#rich-display-rendering - but if I interpret https://github.com/windmill-labs/windmill/blob/d810e178db8810b184a5b030137f1af6d67d0a30/backend/windmill-worker/src/worker.rs#L1368 correctly, it is force-interpreted as string.

can this be changed, or did I do a mistake? :) 

All the best,
Sebastian
PS: I am executing https://playwright.dev/ for making screenshots - this does not exist in deno; so I am resorting to calling node.js via bash.
PPS: Sorry for the many questions in the last days ;) hope thats ok.

**Comment by skurfuerst** [comment, marked as answer]
Nevermind, I work-arounded it by using `Deno.run` instead :)

**Comment by rubenfiszel** [comment]
To give context around this for posterity. The reason why other languages can return json and not bash is that there is no notion of types in bash. In most programming languages, any value is represented by: A Tag, and a memory content or ref to a memory content. There is no such thing in bash, it's just stdout. We could invent our own standard (e.g: if the line start with `json:` then parse what follows. However, I prefer simplicity and not inventing our own standard so I would suggest doing what you already do.

**Reply by skurfuerst** [reply]
Yeah I totally understand. I could imagine some magic like "if it parses as JSON, we interpret it as JSON"; but can totally understand that you prefer less magic.

All the best,
Sebastian

**Comment by jcmcken** [comment]
I know this discussion is considered answered, but it would be nice if the output format of a particular script was simply configurable independent from the contents of the script. I'm not sure how this influences Windmill's architecture though.

Then if I create a Bash script, and I configure Windmill to parse its output as JSON (or YAML or whatever), and the output does not deserialize to JSON (or YAML or whatever), that it simply errors with a server-class error. This way you don't have to come up with some special "magical" protocol for Bash (or whatever other runtime) to emit JSON.
